### PR TITLE
[Voice Broadcast] Update buffering display and improve playback

### PR DESCRIPTION
**Pull request by Florian14**
## Type of change

- [x] WIP Feature
- [ ] Bugfix
- [ ] Technical
- [ ] Other :

## Content

- Move the buffering view to the timeline header
- Display the `pause` button when the voice broadcast is buffering to let the user stop the buffering
- Some UI adjustments (padding, colors)
- Fixed playback issues

## Motivation and context

Design changes:

image

## Tests

1. Start recording a VB from device A
2. Start listening to the VB from device B
3. Observe that the buffering is displayed as expected while waiting for a new chunk
4. Verify that the pause button work as expected during the buffering

## Tested devices

- [ ] Physical
- [x] Emulator
- OS version(s):

## Checklist

- [ ] Changes has been tested on an Android device or Android emulator with API 21
- [ ] UI change has been tested on both light and dark themes
- [ ] Accessibility has been taken into account. See https://github.com/vector-im/element-android/blob/develop/CONTRIBUTING.md#accessibility
- [x] Pull request is based on the develop branch
- [ ] Pull request includes a new file under ./changelog.d. See https://github.com/vector-im/element-android/blob/develop/CONTRIBUTING.md#changelog
- [ ] Pull request includes screenshots or videos if containing UI changes
- [ ] Pull request includes a [sign off](https://matrix-org.github.io/synapse/latest/development/contributing_guide.html#sign-off)
- [x] You've made a self review of your PR
- [ ] If you have modified the screen flow, or added new screens to the application, you have updated the test [UiAllScreensSanityTest.allScreensTest()](https://github.com/vector-im/element-android/blob/main/vector/src/androidTest/java/im/vector/app/ui/UiAllScreensSanityTest.kt#L73)

### CODE REVIEW

File: vector/src/main/java/im/vector/app/features/home/room/detail/timeline/helper/AudioMessagePlaybackTracker.kt [hunk 1]
```diff
@@ -110,21 +111,23 @@ class AudioMessagePlaybackTracker @Inject constructor() {
 
     fun getPlaybackState(id: String) = states[id]
 
-    fun getPlaybackTime(id: String): Int {
+    fun getPlaybackTime(id: String): Int? {
```
Comment (ganfra) [inline vector/src/main/java/im/vector/app/features/home/room/detail/timeline/helper/AudioMessagePlaybackTracker.kt:114]: Why those changes?

File: vector/src/main/java/im/vector/app/features/voicebroadcast/views/VoiceBroadcastBufferingView.kt [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+package im.vector.app.features.voicebroadcast.views
+
+import android.content.Context
+import android.util.AttributeSet
+import android.view.LayoutInflater
+import android.widget.LinearLayout
+import im.vector.app.databinding.ViewVoiceBroadcastBufferingBinding
+
+class VoiceBroadcastBufferingView @JvmOverloads constructor(
```
Comment (ganfra) [inline vector/src/main/java/im/vector/app/features/voicebroadcast/views/VoiceBroadcastBufferingView.kt:25]: There is no binding?

**Review by ganfra — APPROVED** [review]
Some small questions, otherwise LGTM

File: vector/src/main/java/im/vector/app/features/voicebroadcast/views/VoiceBroadcastBufferingView.kt (hunk 2 above)
Comment (Florian14) [reply vector/src/main/java/im/vector/app/features/voicebroadcast/views/VoiceBroadcastBufferingView.kt:25]: Nope, I don't really need to manipulate this custom view from code, it's just for consistency with the other metadata which are icon+text and not progress+text

File: vector/src/main/java/im/vector/app/features/home/room/detail/timeline/helper/AudioMessagePlaybackTracker.kt (hunk 1 above)
Comment (Florian14) [reply vector/src/main/java/im/vector/app/features/home/room/detail/timeline/helper/AudioMessagePlaybackTracker.kt:114]: For Voice Broadcast, I need to distinguish a paused playback at 0sec from another which has no saved position, so I decided to return null in that case. The first time we play a voice broadcast which is live, the user should automatically jump to the last chunk, if he decided to manually scroll in the seekBar, the playback will start from that position (eg. 0 sec)

Comment (ganfra) [reply vector/src/main/java/im/vector/app/features/home/room/detail/timeline/helper/AudioMessagePlaybackTracker.kt:114]: Ok, maybe a sealed class would have been cleaner, but ok!

Comment (Florian14) [reply vector/src/main/java/im/vector/app/features/home/room/detail/timeline/helper/AudioMessagePlaybackTracker.kt:114]: I reused this object, I agree there are several things to improve: having the recording in the states list is a bit weird, same about having a percentage but not the duration, and like you said we don't really need this method and could have it in the sealed class (which already exists)

**Pull request merged**